aws/resource_aws_alb_target_group.go

```diff
@@ -79,6 +80,14 @@ func resourceAwsAlbTargetGroup() *schema.Resource {
 				ValidateFunc: validateAwsAlbTargetGroupDeregistrationDelay,
 			},
 
+			"target_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "instance",
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"ip", "instance"}, true),
```

Comment:
Could you prefer the use of constants (TargetTypeEnumInstance & TargetTypeEnumIp) here?

Reply:
I don't see what extra that would give us? It's for validation of what the user inputs - i'm interested in seeing what you think

Reply:
By using SDK constants everywhere, you rely on the same exact value everywhere. 
Relying on a user-typed string can brings issues if any typo is made. Thus, that's why we try to stick to them as much as possible, whenever they exist.

Reply:
I will update the PR to remove the validation - I don't see any reason why we need to use constants when the values are `ip` or `instance` - if they were longer strings then I would understand

Reply:
Paul,

Perhaps I was misleading in my comment, but the validation was really fine to me.

Was just wondering if we could do that:
```go
ValidateFunc: validation.StringInSlice([]string{elbv2.TargetTypeEnumInstance, elbv2. TargetTypeEnumIp}, true),
```

Sorry if was not clear at all 😅